contrib/mesos/pkg/node/registrator.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	"reflect"
+
+	log "github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+const (
+	defaultInboxSize = 1000
+)
+
+type registration struct {
+	hostName string
+	labels   map[string]string
+}
+
+type Registrator struct {
+	store  cache.Store
+	client *client.Client
+	inbox  chan registration
```

Comment:
may want to consider use of a cache.FIFO so that registration requests for the same host are merged, otherwise the inbox could have multiple entries for the same node, the oldest of which are out of sync with respect to the latest node status (this is not likely, but possible)